server/src/v1/Title/Service/title.service.ts

```diff
@@ -73,34 +72,30 @@ export class TitleService {
         return serializerService.serializeResponse('title_list', result)
     }
 
-    async createTitle(openedBy: string, payload: CreateTitleDto, buffer: Buffer): Promise<ISerializeResponse> {
+    async createTitle(openedBy: string, payload: any, buffer: Buffer): Promise<ISerializeResponse> {
         payload.name = payload.name.replace(/^\s+|\s+$/g, '')
         if (payload.name.length === 0) throw new BadRequestException('Title name can not be whitespace')
+        if (!payload.tags) throw new BadRequestException('tags must be provided')
 
         const dto = new CreateTitleDto()
         dto.name = payload.name
-        dto.categoryId = payload.categoryId
+        dto.tags = payload.tags.split(',')
 
-        return await validate(dto, { validationError: { target: false } }).then(async errors => {
+        const result = await validate(dto, { validationError: { target: false } }).then(async errors => {
             if (errors.length > 0) {
                 throw new BadRequestException(errors)
             }
 
-            let category: CategoriesEntity
-            try {
-                category = await this.categoriesRepository.findOneOrFail(dto.categoryId)
-            } catch (err) {
-                throw new NotFoundException('Category could not found by given id')
-            }
-
-            if (!category.is_leaf) throw new BadRequestException('Category that is not leaf can not have titles')
-
-            const newTitle: TitlesEntity = await this.titlesRepository.createTitle(openedBy, dto, category.ancestors)
+            const newTitle: TitlesEntity = await this.titlesRepository.createTitle(openedBy, dto)
             if (buffer) this.awsService.uploadImage(String(newTitle.id), 'titles', buffer)
 
-            if (configService.isProduction()) sitemapManipulationService.addToIndexedSitemap(newTitle.slug, new Date().toJSON().slice(0,10))
+            if (configService.isProduction()) sitemapManipulationService.addToIndexedSitemap(newTitle.slug, new Date().toJSON().slice(0, 10))
             return serializerService.serializeResponse('title_detail', newTitle)
         })
+
+        dto.tags.forEach(async element => this.tagsRepository.tagActionOnTitleCreateOrUpdate(element))
```

Comment:
For each tag that exists in the title,  it will make a db call to update the db.
- This is inefficient as opening up a db connection is expensive.

We should have a repo function that does a bulk update of the tags in on db connection.